### PR TITLE
Add spring

### PR DESCRIPTION
**Pull request by randrii**

### CODE REVIEW

File: src/main/java/com/rybka/Application.java [hunk 1]
```diff
@@ -1,75 +1,17 @@
...
-
-import java.net.http.HttpClient;
-import java.nio.file.Path;
-import java.nio.file.Paths;
-import java.util.Map;
-import java.util.Scanner;
+import org.springframework.context.annotation.AnnotationConfigApplicationContext;
+import org.springframework.stereotype.Service;
 
+@Service
```
Comment (jarc0der) [inline src/main/java/com/rybka/Application.java]: Why do you use it here?

File: src/main/java/com/rybka/ApplicationConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        return new CVurl();
+    }
+
+    @Bean
+    public CsvMapper csvMapper() {
+        return new CsvMapper();
+    }
+
+    @Bean
+    public Path exportPath() {
```
Comment (jarc0der) [inline src/main/java/com/rybka/ApplicationConfiguration.java]: I believe we don't need bean for this.

File: src/main/java/com/rybka/ApplicationConfiguration.java [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        return Paths.get(environment.getProperty(PropertyInfo.PROPERTY_EXPORT_FOLDER)
+                + FileUtils.generateFileName(environment.getProperty(PropertyInfo.PROPERTY_EXPORT_TYPE)));
+    }
+
+    @Bean
```
Comment (jarc0der) [inline src/main/java/com/rybka/ApplicationConfiguration.java]: Check the @ConditionalOnProperty annotation. It allows spring decide what bean load based on your application properties. In result we can get rid on map here.

File: src/main/java/com/rybka/ApplicationConfiguration.java [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+        return MapSearchUtil.retrieveMapValue(exchangeSourceMap, environment.getProperty(PropertyInfo.PROPERTY_EXCHANGE_SOURCE), new InvalidPropertyException("Unsupported export type or exchange source."));
+    }
+
+    @Bean
+    public Scanner scanner() {
+        return new Scanner(System.in);
+    }
+
+    @Bean
+    public ExchangeService exchangeService(BaseCurrencyExchangeConnector connector) {
```
Comment (jarc0der) [inline src/main/java/com/rybka/ApplicationConfiguration.java]: You don't have to define this Bean here. It should be autoconfigured by Spring. 

Mark ExchangeService with @Service anotation. Spring will find all bean marked with Component -> Service and try to create an instance(bean) for us.

File: src/main/java/com/rybka/config/ExchangeSource.java [hunk 5]
```diff
@@ -1,13 +1,6 @@
 package com.rybka.config;
 
-import lombok.Getter;
-import lombok.RequiredArgsConstructor;
-
-@RequiredArgsConstructor
-@Getter
-public enum ExchangeSource {
```
Comment (jarc0der) [inline src/main/java/com/rybka/config/ExchangeSource.java:8]: Why do you remove enum?

File: src/main/java/com/rybka/view/ExchangeView.java [hunk 6]
```diff
@@ -5,14 +5,18 @@
...
+import javax.annotation.Resource;
 import java.util.Map;
 import java.util.Scanner;
 
+@Component
 @Log4j
 @RequiredArgsConstructor
 public class ExchangeView {
     private final Scanner scanner;
+    @Resource
```
Comment (jarc0der) [inline src/main/java/com/rybka/view/ExchangeView.java]: Why do you use this anot. here?

File: src/main/java/com/rybka/Application.java [hunk 7]
```diff
@@ -1,75 +1,37 @@
...
 public class Application {
 
     public static void main(String[] args) {
 
-        var objectMapper = new ObjectMapper();
+        AnnotationConfigApplicationContext context = new AnnotationConfigApplicationContext(ApplicationConfiguration.class);
 
-        var primeExchangeRateConnector = new PrimeExchangeRateConnector(HttpClient.newHttpClient(), objectMapper);
-        var exchangeRateConnectorConnector = new ExchangeRateConnector(new CVurl());
+        Map<String, BaseCurrencyExchangeConnector> exchangeSourceMap = Map.of(
```
Comment (jarc0der) [inline src/main/java/com/rybka/Application.java]: Try to get rid off the map by using ConditionalOnProperty

https://docs.spring.io/spring-boot/docs/current/api/org/springframework/boot/autoconfigure/condition/ConditionalOnProperty.html

File: src/main/java/com/rybka/Application.java (hunk 1 above)
Comment (randrii) [reply src/main/java/com/rybka/Application.java]: fixed

File: src/main/java/com/rybka/ApplicationConfiguration.java (hunk 2 above)
Comment (randrii) [reply src/main/java/com/rybka/ApplicationConfiguration.java]: fixed

File: src/main/java/com/rybka/view/ExchangeView.java (hunk 6 above)
Comment (randrii) [reply src/main/java/com/rybka/view/ExchangeView.java]: fixed

File: src/main/java/com/rybka/Application.java (hunk 7 above)
Comment (randrii) [reply src/main/java/com/rybka/Application.java:18]: fixed

File: src/main/java/com/rybka/ApplicationConfiguration.java (hunk 4 above)
Comment (randrii) [reply src/main/java/com/rybka/ApplicationConfiguration.java]: fixed

File: src/main/java/com/rybka/ApplicationConfiguration.java (hunk 3 above)
Comment (randrii) [reply src/main/java/com/rybka/ApplicationConfiguration.java]: ok

File: src/main/java/com/rybka/Application.java (hunk 7 above)
Comment (randrii) [reply src/main/java/com/rybka/Application.java]: ok

File: src/main/java/com/rybka/config/ExchangeSource.java (hunk 5 above)
Comment (randrii) [reply src/main/java/com/rybka/config/ExchangeSource.java:8]: enum value isn't constant for component name

**Pull request merged**